libafl/src/executors/inprocess.rs

```diff
@@ -873,6 +874,149 @@ mod unix_signal_handler {
     }
 }
 
+/// Same as `inproc_crash_handler`, but this is called when address sanitizer exits, not from the exception handler
+#[cfg(all(windows, feature = "std"))]
+pub mod windows_asan_handler {
+    use alloc::string::String;
+    use core::{
+        ptr,
+        sync::atomic::{compiler_fence, Ordering},
+    };
+    #[cfg(feature = "std")]
+    use std::io::{stdout, Write};
+
+    use windows::Win32::System::Threading::{
+        EnterCriticalSection, LeaveCriticalSection, RTL_CRITICAL_SECTION,
+    };
+
+    use crate::{
+        corpus::{Corpus, Testcase},
+        events::{Event, EventFirer, EventRestarter},
+        executors::{inprocess::GLOBAL_STATE, Executor, ExitKind, HasObservers},
+        feedbacks::Feedback,
+        fuzzer::HasObjective,
+        inputs::UsesInput,
+        observers::ObserversTuple,
+        state::{HasClientPerfMonitor, HasMetadata, HasSolutions},
+    };
+
+    /// # Safety
+    /// ASAN deatch handler
+    pub unsafe extern "C" fn asan_death_handler<E, EM, OF, Z>()
+    where
+        E: Executor<EM, Z> + HasObservers,
+        EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
+        OF: Feedback<E::State>,
+        E::State: HasSolutions + HasClientPerfMonitor,
+        Z: HasObjective<Objective = OF, State = E::State>,
+    {
+        let mut data = &mut GLOBAL_STATE;
```

Comment:
they can't have a single function because they have different signatures.
but I can group up common code of these handlers and make them into another subroutine. 

but I want to ask why we keep this mod windows_exception_handler private?
can we make it public?

Reply:
Why should it be part of the API? Should a user ever be able to call it directly? If not, it should be private, or public(crate)

Reply:
Right now users are intended to call it manually. But we might just call it regardless of whether asan is enabled or not.

Reply:
libafl_targets need to use the asan_death_callback, that is another crate. So I must make it `asan_death_callback` public, but I can make the internals into several functions and hide them

Reply:
Would be cool to deduplicate code where possible